tests/testflows/map_type/requirements/requirements.md

```diff
@@ -0,0 +1,523 @@
+    * 3.13.6 [`mapContains`](#mapcontains)
+      * 3.13.6.1 [RQ.SRS-018.ClickHouse.Map.DataType.Functions.MapContains](#rqsrs-018clickhousemapdatatypefunctionsmapcontains)
+    * 3.13.7 [`mapKeys`](#mapkeys)
+      * 3.13.7.1 [RQ.SRS-018.ClickHouse.Map.DataType.Functions.MapKeys](#rqsrs-018clickhousemapdatatypefunctionsmapkeys)
+    * 3.13.8 [`mapValues`](#mapvalues)
+      * 3.13.8.1 [RQ.SRS-018.ClickHouse.Map.DataType.Functions.MapValues](#rqsrs-018clickhousemapdatatypefunctionsmapvalues)
+
+## Revision History
+
+This document is stored in an electronic form using [Git] source control management software
+hosted in a [GitHub Repository].
+All the updates are tracked using the [Revision History].
+
+## Introduction
+
+This software requirements specification covers requirements for `Map(key, value)` data type in [ClickHouse].
+
+## Requirements
+
+### General
+
+#### RQ.SRS-018.ClickHouse.Map.DataType
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type that stores `key:value` pairs.
+
+### Performance
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Performance.Vs.ArrayOfTuples
+version:1.0
+
+[ClickHouse] SHALL provide comparable performance for `Map(key, value)` data type as
+compared to `Array(Tuple(K,V))` data type.
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Performance.Vs.TupleOfArrays
+version:1.0
+
+[ClickHouse] SHALL provide comparable performance for `Map(key, value)` data type as
+compared to `Tuple(Array(String), Array(String))` data type where the first
+array defines an array of keys and the second array defines an array of values.
+
+### Key Types
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Key.String
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type where key is of a [String] type.
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Key.Integer
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type where key is of an [Integer] type.
+
+### Value Types
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Value.String
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type where value is of a [String] type.
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Value.Integer
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type where value is of a [Integer] type.
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Value.Array
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type where value is of a [Array] type.
+
+### Invalid Types
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Invalid.Nullable
+version: 1.0
+
+[ClickHouse] SHALL not support creating table columns that have `Nullable(Map(key, value))` data type.
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.Invalid.NothingNothing
+version: 1.0
+
+[ClickHouse] SHALL not support creating table columns that have `Map(Nothing, Nothing))` data type.
+
+### Duplicated Keys
+
+#### RQ.SRS-018.ClickHouse.Map.DataType.DuplicatedKeys
+version: 1.0
+
+[ClickHouse] SHALL support `Map(key, value)` data type with duplicated keys.
```

Comment:
Not "SHALL".

Reply:
You mean Map type shall **not** support duplicated keys?

Reply:
No.

It's unspecified behaviour.
We cannot say it should support duplicate keys or not.

Reply:
The SRS just specifies and documents current behavior that is verified by tests. If we want to keep this unspecified it is best to mention this in the docs. I propose we leave this requirement as is.

Reply:
Updated the requirement to use MAY.